### ISSUE TITLE
New analyzer: Collection properties should not make copies

### ISSUE THREAD

**Issue by Evangelink** [dotnet/roslyn-analyzers#3569]
When calling a property, developers expect the call to be as efficient as a field access. However if the property returns a copy of a collection it might be much slower than a field access. This won't be the expected behavior.

We could start by detecting calls to `ToList`, `ToArray` and `Clone`.

```csharp
public class C
{
	public C[] ArrayOfC => (C[])_myArray.Clone();
	public IEnumerable<C> Enumerable => _myList.ToList();
}
```

**Comment by Evangelink** [comment]
@mavasani Could you confirm you are fine with this rule?
